diff --git b/gcc/fortran/module.c a/gcc/fortran/module.c
index f0f8f97..677bb32 100644
--- b/gcc/fortran/module.c
+++ a/gcc/fortran/module.c
@@ -2550,7 +2550,7 @@ mio_component_ref (gfc_component **cp, gfc_symbol *sym)
       if (sym->components != NULL && p->u.pointer == NULL)
 	{
 	  /* Symbol already loaded, so search by name.  */
-	  q = gfc_find_component (sym, name, true, true);
+	  q = gfc_find_component (sym, name, true, true, NULL);
 
 	  if (q)
 	    associate_integer_pointer (p, q);
