From d222e2005bbdd5dead5423d91dd85cbace687938 Mon Sep 17 00:00:00 2001
From: Jim MacArthur <jim.macarthur@codethink.co.uk>
Date: Thu, 4 Feb 2016 16:59:41 +0000
Subject: [PATCH 26/46] Allow blank format items in format strings

---
 gcc/fortran/io.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/gcc/fortran/io.c b/gcc/fortran/io.c
index 1aa306d..0bfb20c 100644
--- a/gcc/fortran/io.c
+++ b/gcc/fortran/io.c
@@ -709,6 +709,11 @@ format_item_1:
       error = unexpected_end;
       goto syntax;
 
+    case FMT_RPAREN:
+      /* Oracle allows a blank format item. If this flag is off,
+	 fall through to default. */
+      if (gfc_option.allow_std & GFC_STD_EXTRA_LEGACY) goto finished;
+
     default:
       error = unexpected_element;
       goto syntax;
-- 
1.9.1

